steering_controllers_library/src/steering_odometry.cpp

```diff
@@ -0,0 +1,341 @@
+
+  if (Vx == 0 && theta_dot != 0)
+  {
+    alpha = theta_dot > 0 ? M_PI_2 : -M_PI_2;
+    Ws = abs(theta_dot) * wheelbase_ / wheel_radius_;
+  }
+  else
+  {
+    alpha = SteeringOdometry::convert_trans_rot_vel_to_steering_angle(Vx, theta_dot);
+    Ws = Vx / (wheel_radius_ * std::cos(steer_pos_));
+  }
+
+  if (config_type_ == BICYCLE_CONFIG)
+  {
+    std::vector<double> traction_commands = {Ws};
+    std::vector<double> steering_commands = {alpha};
+    return std::make_tuple(traction_commands, steering_commands);
+  }
+  else if (config_type_ == TRICYCLE_CONFIG)
+  {
+    std::vector<double> traction_commands;
+    std::vector<double> steering_commands;
+    if (fabs(steer_pos_) < 1e-6)
+    {
+      traction_commands = {Ws, Ws};
+    }
+    else
+    {
+      double turning_radius = wheelbase_ / std::tan(steer_pos_);
+      double Wr = Ws * (turning_radius + wheel_track_ * 0.5) / turning_radius;
+      double Wl = Ws * (turning_radius - wheel_track_ * 0.5) / turning_radius;
+      traction_commands = {Wr, Wl};
+    }
+    steering_commands = {alpha};
+    return std::make_tuple(traction_commands, steering_commands);
+  }
+  else if (config_type_ == ACKERMANN_CONFIG)
+  {
+    std::vector<double> traction_commands;
+    std::vector<double> steering_commands;
+    if (fabs(steer_pos_) < 1e-6)
+    {
+      traction_commands = {Ws, Ws};
+      steering_commands = {alpha, alpha};
+    }
+    else
+    {
+      double turning_radius = wheelbase_ / std::tan(steer_pos_);
+      double Wr = Ws * (turning_radius + wheel_track_ * 0.5) / turning_radius;
+      double Wl = Ws * (turning_radius - wheel_track_ * 0.5) / turning_radius;
+      traction_commands = {Wr, Wl};
+
+      double numerator = 2 * wheelbase_ * std::sin(alpha);
+      double denominator_first_member = 2 * wheelbase_ * std::cos(alpha);
+      double denominator_second_member = wheel_track_ * std::sin(alpha);
+
+      double alpha_r = std::atan2(numerator, denominator_first_member - denominator_second_member);
+      double alpha_l = std::atan2(numerator, denominator_first_member + denominator_second_member);
+      steering_commands = {alpha_r, alpha_l};
+    }
+    return std::make_tuple(traction_commands, steering_commands);
+  }
+  else
+  {
+    throw std::runtime_error("Config not implemented");
+  }
+}
+
+void SteeringOdometry::reset_odometry()
+{
+  x_ = 0.0;
+  y_ = 0.0;
+  heading_ = 0.0;
+  reset_accumulators();
+}
+
+void SteeringOdometry::integrate_runge_kutta_2(double linear, double angular)
+{
+  const double direction = heading_ + angular * 0.5;
+
+  /// Runge-Kutta 2nd order integration:
+  x_ += linear * cos(direction);
+  y_ += linear * sin(direction);
+  heading_ += angular;
+}
+
+/**
+   * \brief Other possible integration method provided by the class
+   * \param linear
+   * \param angular
+   */
+void SteeringOdometry::integrate_exact(double linear, double angular)
+{
+  if (fabs(angular) < 1e-6)
+    integrate_runge_kutta_2(linear, angular);
```

Comment:
```suggestion
{
    integrate_runge_kutta_2(linear, angular);
}
```

Because I'm not sure if the Format job will eat this without brackets